webui/react/src/hooks/useSettings.ts

```diff
@@ -271,27 +281,41 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
 
         newSettings[setting as keyof T] = defaultSetting.defaultValue;
       });
+      stateOb.update((s) => s.set(config.storagePath, newSettings));
 
-      update(config.storagePath, newSettings);
-
-      await updateDB(newSettings);
+      await updateDB(newSettings, user);
 
       navigate('', { replace: true });
     },
-    [config, update, updateDB, navigate, settings],
+    [config, updateDB, navigate, settings, user, stateOb],
   );
 
   const updateSettings = useCallback(
-    async (updates: Settings, shouldPush = false) => {
-      if (!settings) return;
-
-      const newSettings = { ...settings, ...updates };
-
-      if (isEqual(newSettings, settings)) return;
-
-      update(config.storagePath, newSettings);
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    (updates: Settings, shouldPush = false) => {
```

Comment:
Since we switched to `subscribe`, not sure what should be the best way to pass `shouldPush` parameter here